components/support/base/src/main/java/mozilla/components/support/base/observer/ObserverRegistry.kt

```diff
@@ -35,17 +39,18 @@ class ObserverRegistry<T> : Observable<T> {
     }
 
     override fun register(observer: T, owner: LifecycleOwner, autoPause: Boolean) {
-        if (owner.lifecycle.currentState == Lifecycle.State.DESTROYED) {
+        // Don't register if the owner is already destroyed
+        if (owner.lifecycle.currentState == DESTROYED) {
             return
         }
 
         register(observer)
 
-        val lifecycleObserver = LifecycleBoundObserver(
-                owner,
-                registry = this,
-                observer = observer,
-                autoPause = autoPause)
+        val lifecycleObserver = if (autoPause) {
+            AutoPauseLifecycleBoundObserver(owner, registry = this, observer = observer)
+        } else {
+            LifecycleBoundObserver(owner, registry = this, observer = observer)
+        }
```

Comment:
If we specify `autoPause` then we do not have the destroy behavior here anymore, or am I missing something?

Reply:
`AutoPauseLifecycleBoundObserver` extends `LifecycleBoundObserver`

Reply:
Oops, missed that, somehow I thought it extended `LifecycleObserver`.